core/impl/src/main/java/pt/webdetails/cdf/dd/model/meta/reader/cdexml/XmlComponentTypeReader.java

```diff
@@ -184,11 +185,11 @@ private String readBaseProperties( ComponentType.Builder builder, Element elem,
     }
 
     @SuppressWarnings( "unchecked" )
-    List<Element> legacyNamesElems = elem.selectNodes( "Header/LegacyIName" );
-    for ( Element legacyNameElem : legacyNamesElems ) {
-      String legacyName = legacyNameElem.getStringValue();
-      if ( StringUtils.isNotBlank( legacyName ) ) {
-        builder.addLegacyName( legacyName );
+    List<Node> nodes = elem.selectNodes( "Header/LegacyIName" );
```

Comment:
Why would you change `Element` to Node, if we're actually selecting elements? To avoid a dynamic cast? Should you not, at least, remove the "unchecked" then?
You can do this if we're not really depending on `Element` specific members, though.

However, I disagree with reducing all names to `nodes`. There was useful meaning to the name `legacyNamesElems`. Why not `legacyNamesNodes` or `legacyNames` instead?

Reply:
Because in dom4j 2.x, `selectNodes()` will always return a list of `Node`, not `Element`.